search/mongo_search.go

```diff
@@ -596,24 +694,10 @@ func (m *MongoSearcher) createReferenceQueryObject(r *ReferenceParam) bson.M {
 			}
 		case ExternalReference:
 			criteria["reference"] = ci(ref.URL)
+
 		case ChainedQueryReference:
-			// Since MongoDB does not support cross-collection searches, we must break this into two:
-			// (1) perform search against referenced collection using chained search Query
-			// (2) use ID results from first query to build second query
-			// TODO: Investigate if new Mongo 3.2 $lookup pipeline feature might be an improvement
-			var idObjs []struct {
-				ID string `bson:"_id"`
-			}
-			q := m.CreateQueryWithoutOptions(ref.ChainedQuery)
-			q.Select(bson.M{"_id": 1}).All(&idObjs)
-			ids := make([]string, len(idObjs))
-			for i := range idObjs {
-				ids[i] = idObjs[i].ID
-			}
-			criteria["referenceid"] = bson.M{"$in": ids}
-			if ref.Type != "" {
-				criteria["type"] = ref.Type
-			}
+			// This should be handled exclusively by the createPipelineObject
```

Comment:
What about the chained queries that are on embedded fields (e.g., Bundle's `message` parameter)?  Don't those still need this or is that handled elsewhere?